src/pathfinding_igvc/include/OccupancyGridResizeService.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Created By: Min Gyo Kim
+ * Created On: May 14th 2018
+ * Description: A service that enlarges the occupancy grid to fit the goal point
+ * if it's outside of the grid
+ */
+
+#ifndef PATHFINDING_IGVC_OCCUPANCYGRIDRESIZESERVICE_H
+#define PATHFINDING_IGVC_OCCUPANCYGRIDRESIZESERVICE_H
+
+#include "AStar.h"
+
+class OccupancyGridResizeService {
+  public:
+    /**
+     * Given an occupancy grid and a cell location, resize the grid if the cell
+     * location is not
+     * inside the grid (i.e. row/col is < 0 or >= width/height).
+     * The grid will also move its origin if the cell location is to the left or
+     * below
+     * the initial origin of the grid.
+     * If the cell location is already inside the grid, it doesn't change the
+     * grid
+     *
+     * @param grid the occupancy grid passed by reference
+     * @param goal the cell location that we want to fit inside the grid
+     */
+    void static resizeOccupancyGridToFitGoal(nav_msgs::OccupancyGrid& grid,
+                                             AStar::GridPoint goal);
```

Comment:
This seems like it might be a very expensive operation. Keep in mind we might have a very high resolution graph (say 10cm per point) that's 5m by 5m (25m^2) and the goal might be 50m away, in which case an occupancy grid of that size would probably use GB's of memory and take a while to build. 

My suggestion for this would be to increase the size of the occupancy grid by one in every direction (with all added grid cells being unoccupied) then make the goal node the  closest node in these additional nodes the closest node in these newly added nodes to the goal node
![screenshot from 2018-05-16 21-57-48](https://user-images.githubusercontent.com/9075711/40157490-31ae8d00-5954-11e8-9b38-a1d0a21431c9.png)


Let me know what you think, this is definitely a very tricky problem, but I _think_ this solution should be fairly simple to implement?

Reply:
Yeah that would be much more efficient. Sounds easy to implement as well